pkg/webhook/validatingwebhook.go

```diff
@@ -0,0 +1,184 @@
+	errorMsg string
+	// apiClient opens api connection to the cluster.
+	apiClient *clients.Settings
+}
+
+// PullValidatingConfiguration pulls existing ValidatingWebhookConfiguration from cluster.
+func PullValidatingConfiguration(apiClient *clients.Settings, name string) (
+	*ValidatingConfigurationBuilder, error) {
+	glog.V(100).Infof("Pulling existing ValidatingWebhookConfiguration name %s from cluster", name)
+
+	builder := ValidatingConfigurationBuilder{
+		apiClient: apiClient,
+		Definition: &admregv1.ValidatingWebhookConfiguration{
+			ObjectMeta: metaV1.ObjectMeta{
+				Name: name,
+			},
+		},
+	}
+
+	if name == "" {
+		glog.V(100).Infof("The name of the ValidatingWebhookConfiguration is empty")
+
+		builder.errorMsg = "ValidatingWebhookConfiguration 'name' cannot be empty"
+	}
+
+	if !builder.Exists() {
+		return nil, fmt.Errorf("ValidatingWebhookConfiguration object %s doesn't exist", name)
+	}
+
+	builder.Definition = builder.Object
+
+	return &builder, nil
+}
+
+// Exists checks whether the given ValidatingWebhookConfiguration object exists in the cluster.
+func (builder *ValidatingConfigurationBuilder) Exists() bool {
+	if valid, _ := builder.validate(); !valid {
+		return false
+	}
+
+	var err error
+	builder.Object, err = builder.Get()
+
+	return err == nil || !k8serrors.IsNotFound(err)
+}
+
+// Get returns ValidatingWebhookConfiguration object.
+func (builder *ValidatingConfigurationBuilder) Get() (*admregv1.ValidatingWebhookConfiguration, error) {
+	if valid, err := builder.validate(); !valid {
+		return &admregv1.ValidatingWebhookConfiguration{}, err
+	}
+
+	validatingWebhookConfiguration := &admregv1.ValidatingWebhookConfiguration{}
+	err := builder.apiClient.Get(context.TODO(), goclient.ObjectKey{
+		Name: builder.Definition.Name,
+	}, validatingWebhookConfiguration)
+
+	if err != nil {
+		glog.V(100).Infof("Failed to get ValidatingWebhookConfiguration %s: %v", builder.Definition.Name, err)
+
+		return &admregv1.ValidatingWebhookConfiguration{}, err
+	}
+
+	return validatingWebhookConfiguration, err
+}
+
+// Delete removes a ValidatingWebhookConfiguration from a cluster.
+func (builder *ValidatingConfigurationBuilder) Delete() (*ValidatingConfigurationBuilder, error) {
+	if valid, err := builder.validate(); !valid {
+		return builder, err
+	}
+
+	glog.V(100).Infof("Deleting the ValidatingWebhookConfiguration %s", builder.Definition.Name)
+
+	if !builder.Exists() {
+		return builder, fmt.Errorf("ValidatingWebhookConfiguration cannot be deleted because it does not exist")
+	}
+
+	err := builder.apiClient.Delete(context.TODO(), builder.Definition)
+
+	if err != nil {
+		return builder, fmt.Errorf("can not delete ValidatingWebhookConfiguration: %w", err)
+	}
+
+	builder.Object = nil
+
+	return builder, nil
+}
+
+// DeleteAndWait deletes an ValidatingWebhookConfiguration and waits until it is removed from the cluster.
+func (builder *ValidatingConfigurationBuilder) DeleteAndWait(timeout time.Duration) error {
```

Comment:
I don't think DeleteAndWait is relevant for this object. It removes immediately from cluster and doesn't have statuses like "Terminating" or "Pending".

Reply:
removed